test/runtest.py

```diff
@@ -1511,6 +1511,26 @@ def wait_for_object_table():
 
     ray.worker.cleanup()
 
+  def testTaskProfileAPI(self):
+    ray.init(redirect_output=True)
+
+    @ray.remote
+    def say_hi():
+      print("hi")
+
+    say_hi.remote()
+
+    # Make sure that each task executes
+    start_time = time.time()
+    while time.time() - start_time < 10:
+      profiles = ray.global_state.task_profiles()
+      profile_str = "".join(str(x) for x in profiles)
```

Comment:
Let's test this in a slightly different way. Let's not make this into a giant string, but instead check that it has one key (corresponding to the task ID) and that the corresponding value has keys corresponding to the things that we expect (e.g., keys for the time required for execution and storing stuff in the object store)